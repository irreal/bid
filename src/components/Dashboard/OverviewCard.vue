<template>
  <div class="overviewCard" @click="$emit('click')">
    <div class="overviewCard-icon overviewCard-icon--document">
      <i class="fas" :class="{ [icon]: icon }"></i>
    </div>
    <div class="overviewCard-description">
      <h3 class="overviewCard-title text-light">
        {{ text }}
        <strong>{{ textHighlight }}</strong>
      </h3>
      <p class="overviewCard-subtitle">{{ subtitle }}</p>
    </div>
  </div>
</template>

<style scoped lang="scss">
@import "./src/styles/colors.scss";

.overviewCard {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 12px;
  background-color: $color-white;
  transform: translateY(0);
  transition: all 0.3s;

  &-icon {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 60px;
    width: 60px;
    border-radius: 50%;
    font-size: 21px;
    color: #fff;

    &--document {
      background-color: $color-zest;
    }

    &--calendar {
      background-color: $color-jungle-green;
    }

    &--mail {
      background-color: $color-cinnabar;
    }

    &--photo {
      background-color: $color-amethyst;
    }
  }

  &-description {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  &-title {
    font-size: 18px;
    color: $color-java;
    margin: 0;
  }

  &-subtitle {
    margin: 2px;
    color: $color-abbey;
  }

  &:hover {
    transform: translateY(-3px);
    box-shadow: 0 5px 5px rgba(0, 0, 0, 0.1);
    cursor: pointer;
  }
}
</style>

<script>
export default {
  name: "Welcome",
  props: {
    icon: String,
    text: String,
    textHighlight: String,
    subtitle: String
  }
};
</script>
