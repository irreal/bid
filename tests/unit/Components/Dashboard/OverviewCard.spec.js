import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import OverviewCard from "@/components/Dashboard/OverviewCard";
import Vuetify from "vuetify";
import Vue from "vue";

const localVue = createLocalVue();
localVue.use(Vuex);
Vue.use(Vuetify);

describe("Overview Card", () => {
  it("renders the component", () => {
    // arrange
    const card = shallowMount(OverviewCard);
    // assert
    expect(card.html()).toMatchSnapshot();
  });

  it("renders the component with slots", () => {
    // arrange
    const card = shallowMount(OverviewCard, {
      slots: {
        header: "<p>this is a header</p>"
      }
    });
    // assert
    expect(card.html()).toMatchSnapshot();
    expect(card.html()).toContain("this is a header");
  });
});
