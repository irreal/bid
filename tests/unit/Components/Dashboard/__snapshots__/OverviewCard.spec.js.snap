// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Card renders the component 1`] = `
<v-card-stub tag="div">
  <!---->
  <v-card-text-stub></v-card-text-stub>
  <!---->
  <!---->
</v-card-stub>
`;

exports[`Overview Card renders the component with slots 1`] = `
<v-card-stub tag="div" style="margin-bottom: 24px; margin-top: 48px;">
  <helper-offset-stub offset="24">
    <v-card-stub tag="div" color="secondary" dark="true" class="v-card--material__header elevation-10">
      <p>this is a header</p>
    </v-card-stub>
  </helper-offset-stub>
  <v-card-text-stub></v-card-text-stub>
  <!---->
  <!---->
</v-card-stub>
`;
